application_repo/application_repo.go

```diff
@@ -0,0 +1,45 @@
+package application_repo
+
+import "github.com/cloudfoundry/cli/plugin"
+
+//ApplicationRepo - cli connection wrapper
+type ApplicationRepo struct {
+	conn plugin.CliConnection
+}
+
+//NewApplicationRepo - constructor function to create cli connection wrapper
+func NewApplicationRepo(conn plugin.CliConnection) *ApplicationRepo {
+	return &ApplicationRepo{
+		conn: conn,
+	}
+}
+
+//RenameApplication - rename the application given
+func (repo *ApplicationRepo) RenameApplication(oldName, newName string) error {
+	_, err := repo.conn.CliCommand("rename", oldName, newName)
+	return err
+}
+
+//PushApplication - push the application to cf
+func (repo *ApplicationRepo) PushApplication(args []string) error {
+	_, err := repo.conn.CliCommand(args...)
+	return err
+}
+
+//DeleteApplication - delete the application from cf
+func (repo *ApplicationRepo) DeleteApplication(appName string) error {
+	_, err := repo.conn.CliCommand("delete", appName, "-f")
+	return err
+}
+
+//ListApplications - list applications on cf
+func (repo *ApplicationRepo) ListApplications() error {
```

Comment:
i think these existed previously, but we might consider removing them in favor of 
`GetApps()` in our repo.conn object
https://github.com/cloudfoundry/cli/blob/master/plugin/cli_connection.go#L312